src/Nethermind/Nethermind.Blockchain/Producers/BlockProducerBase.cs

```diff
@@ -278,20 +278,20 @@ private IEnumerable<Transaction> GetTransactions(BlockHeader parent)
             return _txSource.GetTransactions(parent, gasLimit);
         }
 
-        protected virtual Block PrepareBlock(BlockHeader parent)
+        protected virtual Block PrepareBlock(BlockHeader parent, Address? blockAuthor = null, UInt256? manualTimestamp = null)
         {
             UInt256 timestamp = UInt256.Max(parent.Timestamp + 1, Timestamper.UnixTime.Seconds);
             BlockHeader header = new(
                 parent.Hash!,
                 Keccak.OfAnEmptySequenceRlp,
-                Sealer.Address,
+                blockAuthor ?? Sealer.Address,
```

Comment:
should there be no PosSealer that would pick the block author properly?

Reply:
@kristofgazso